test/unit/lookup_key_test.rb

```diff
@@ -24,7 +24,7 @@ def test_path2match_single_domain_path
     key   = ""
     value = ""
     as_admin do
-      key   = LookupKey.create!(:key => "ntp", :path => "domain", :puppetclass => Puppetclass.first)
+      key   = LookupKey.create!(:key => "ntp", :path => "domain", :puppetclass => Puppetclass.first, :key_type => "string")
```

Comment:
respond_to? was a reference to the code you removed in this PR at https://github.com/theforeman/foreman/pull/2399/files#diff-5ae27528346794f59c60bcc7cfa4acb4L153, which previously would have skipped casting for a nil/blank key_type.

The caster seems to run when the key_type is nil:

<pre>
2.0.0-p353 :001 > LookupKey.new(:key => "ntp", :path => "domain", :puppetclass => Puppetclass.first, :default_value => 'a').valid?
2015-08-12T10:28:41 [app] [W] Unable to type cast a to 
 => false 
</pre>

However on develop:

<pre>
2.0.0-p353 :001 > LookupKey.new(:key => "ntp", :path => "domain", :puppetclass => Puppetclass.first, :default_value => 'a').valid?
 => true 
</pre>

Which I guess might be why you've set key_type explicitly on all of these tests.